scripts/Engineering/texture/TextureUtils/fitpeaks_engine.py

```
@@ -0,0 +1,485 @@
+            refine = valid & np.isfinite(x0_pass) & (x0_pass > tof_lo) & (x0_pass < tof_hi)
+            centre_seed[refine] = x0_pass[refine]
+            return centre_seed
+
+        def _build_seed_table(valid: np.ndarray) -> str:
+            """Build the per-spectrum PeakParameterValueTable for the next pass from the pass just
+            run (param_tab_name).  One column per carried shape parameter and one row per spectrum in
+            workspace-index order (FitPeaks maps table row -> workspace index).  A spectrum whose fit
+            was not valid (or a non-finite fitted value) is written as NaN, which FitPeaks reads as
+            'no seed for this spectrum/parameter' and leaves at the value from the instrument
+            parameters - so a failed guiding fit never poisons the next pass's shape seed."""
+            src = ADS.retrieve(param_tab_name)
```

Ruff (0.16.1)

[warning] Function definition does not bind loop variable `param_tab_name`

(B023)